src/feed.test.tsx

```diff
@@ -1,54 +1,69 @@
 import React from 'react';
 import { shallow } from 'enzyme';
 import { FeedItem, Model as FeedItemModel } from './feed-item';
-import { Feed, Properties } from './feed';
+import { Feed, Properties, State } from './feed';
 
 describe('Feed', () => {
-  const subject = (props: Partial<Properties>) => {
+  const subject = (props: Partial<Properties>, state: Partial<State>) => {
     const allProps: Properties = {
       items: [],
       app: '',
       ...props,
     };
 
-    return shallow(<Feed {...allProps} />);
+    const allState: State = {
+      feed: [],
+      ...state,
+    };
+
+    return shallow(<Feed {...allProps} {...allState} />);
   };
 
   test('renders a feed item for item', () => {
-    const items = [{
-      id: 'the-first-id',
-      title: 'The First Item',
-      description: 'This is the description of the first item.',
-    }, {
-      id: 'the-second-id',
-      title: 'The Second Item',
-      description: 'This is the description of the Second item.',
-    }] as FeedItemModel[];
-
-    const wrapper = subject({ items });
+    const items = [
+      {
+        id: 'the-first-id',
+        title: 'The First Item',
+        description: 'This is the description of the first item.',
+      },
+      {
+        id: 'the-second-id',
+        title: 'The Second Item',
+        description: 'This is the description of the Second item.',
+      },
+    ] as FeedItemModel[];
+
+    const feed = items;
+
+    const wrapper = subject({ items }, { feed });
 
     expect(wrapper.find(FeedItem)).toHaveLength(2);
   });
 
   test('passes item props to FeedItem', () => {
-    const items = [{
-      id: 'the-first-id',
-      title: 'The First Item',
-      description: 'This is the description of the first item.',
-    }, {
-      id: 'the-second-id',
-      title: 'The Second Item',
-      description: 'This is the description of the Second item.',
-    }] as FeedItemModel[];
-
-    const wrapper = subject({ items });
+    const items = [
+      {
+        id: 'the-first-id',
+        title: 'The First Item',
+        description: 'This is the description of the first item.',
+      },
+      {
+        id: 'the-second-id',
+        title: 'The Second Item',
+        description: 'This is the description of the Second item.',
+      },
+    ] as FeedItemModel[];
+
+    const feed = items;
+
+    const wrapper = subject({ items }, { feed });
 
     expect(wrapper.find(FeedItem).at(1).props()).toEqual(
       expect.objectContaining({
         id: 'the-second-id',
         title: 'The Second Item',
         description: 'This is the description of the Second item.',
-      }),
+      })
```

Comment:
we've added some significant business logic with this. it would be good to see tests that demonstrate the new stuff we added. we should also update most of the tests to include more items than would fit on a page. as well as a test with less items than fit on a page, to make sure nothing weird is going on.

eg. .. we should at least make sure to test that:
- we're rendering the correct items initially.
- we render the correct items on scroll
- that we maintain the correct scroll position, and rendered items, if items are added to the end of the array
- that we maintain the correct scroll position, and render the correct data if one of the items has a property updated.